src/components/molecules/TableHeader/TableHeader.tsx

```diff
@@ -149,10 +168,8 @@ const TableHeader: React.FC<TableHeaderProps> = ({
           <label className="switch">
             <input
               type="checkbox"
-              checked={!!tableLocked(seatedAtTable)}
-              onChange={() =>
-                onLockedChanged(seatedAtTable, !tableLocked(seatedAtTable))
-              }
+              checked={!!isCurrentTableLocked}
+              onChange={() => setIsCurrentTableLocked(!isCurrentTableLocked)}
```

Comment:
You can call `onChange={setIsCurrentTableLocked}` directly and use `isCurrentTableLocked` without passing it as a parameter because it's defined at the top of the component.

Reply:
I don't think I can, since `setIsCurrentTableLocked` is not a toggle fn. It is used as `setIsCurrentTableLocked(false)` as well

Reply:
Just noticed that you need it for the `useEfffect` but in this case `onChange` won't memo the function because of the arrow syntax. Maybe you can make the parameter optional.